packages/contracts/test/attestation.test.ts

```diff
@@ -1,42 +1,95 @@
 // @ts-ignore
 import { ethers } from 'hardhat'
 import { expect } from 'chai'
-import { hash4, SparseMerkleTree } from '@unirep/crypto'
+
+import { AggregateEpochKeysProof, EPOCH_LENGTH } from '../src'
+import { deployUnirep } from '../deploy'
+import {
+    genRandomSalt,
+    hash3,
+    hash4,
+    hash6,
+    SparseMerkleTree,
+    stringifyBigInts,
+} from '@unirep/crypto'
 import {
+    AGGREGATE_KEY_COUNT,
+    Circuit,
     EPOCH_TREE_DEPTH,
-    STATE_TREE_DEPTH,
-    NUM_EPOCH_KEY_NONCE_PER_EPOCH,
+    defaultEpochTreeLeaf,
 } from '@unirep/circuits'
+import { defaultProver } from '@unirep/circuits/provers/defaultProver'
 
-import { EPOCH_LENGTH } from '../src'
-import { deployUnirep } from '../deploy'
+function genCircuitInputs(
```

Comment:
can this be renamed to something more descriptive? It's unclear what circuit the inputs are for.

Reply:
[80c4fc2](https://github.com/Unirep/Unirep/pull/187/commits/80c4fc2fe5a6a850f319ab794d13a070b3129531)